lib/AST/ASTPrinter.cpp

```diff
@@ -3814,6 +3828,13 @@ class TypePrinter : public TypeVisitor<TypePrinter> {
         break;
       case SILFunctionType::Representation::CFunctionPointer:
         Printer << "c";
+        // FIXME: [clang-function-type-serialization] Once we start serializing
+        // Clang function types, we should be able to remove the second check.
+        if (printNameOnly || !info.getUncommonInfo().hasValue())
+          break;
+        Printer << ", cType: "
+                << info.getUncommonInfo().getValue().printClangFunctionType(
+                    Ctx.getClangModuleLoader(), buf);
```

Comment:
You can pretty easily extract a common function in this file that prints the double-quoted cType string.

Reply:
That would need to be templated because the two `ExtInfo`s and the two `Uncommon`s are different types. But yeah, I can do it.

Reply:
Well, you could just pass it a type, a context, and a stream.  I was just thinking that my request to move the quoting out here was going to make this part relatively boilerplate-y, so it'd be nice to have it in one place even if the calling code has some repetition.

Reply:
But we don't have the type in hand, that's the thing. The field is private (well, the subclasses are friends so they have access) and there are no getters; I only added a public method to print the type. I could add a getter...

Or we could have it templated, that's fine, it is only getting instantiated 2 times.

Reply:
Sure, however you like.